ext/crates/fp/tests/row_reduce.rs

```diff
@@ -0,0 +1,73 @@
+use proptest::prelude::*;
+
+use fp::{matrix::Matrix, prime::ValidPrime};
+
+fn arb_prime() -> impl Strategy<Value = ValidPrime> {
+    prop_oneof![
+        Just(ValidPrime::new(2)),
+        Just(ValidPrime::new(3)),
+        Just(ValidPrime::new(5)),
+        Just(ValidPrime::new(7)),
+    ]
+}
+
+fn arb_tableau(rows: usize, cols: usize) -> impl Strategy<Value = Vec<usize>> {
```

Comment:
Could you add a comment explaining the logic here? It's not completely obvious. Should we just assert `rows <= cols`? Arguably it doesn't make much sense otherwise.